src/heretic/model.py

```diff
@@ -227,16 +227,26 @@ def abliterate(
 
                 # Projects any right-multiplied vector(s) onto the subspace
                 # spanned by the refusal direction.
-                projector = torch.outer(
-                    layer_refusal_direction,
-                    layer_refusal_direction,
-                ).to(self.model.dtype)
+                # We use the property (r r^T) W = r (r^T W) to avoid computing
+                # the O(d^2 k) projector matrix and the O(d^3) matrix multiplication.
```

Comment:
I think those complexities are still not correct. Computing the projector takes $d^2$ float multiplications, not $d^2 k$, and multiplying it with the matrix takes $d^2 k$ multiplications, not $d^3$.